### PR TITLE
Unlock php 8 + fixes for symfony 5

### PR DESCRIPTION
**Pull request by rabauss**
Allow to use the bundle with php 8 and fix symfony 5 support.
I'm not sure if we should drop the support for contao 4.4 - what do you think?

### CODE REVIEW

File: src/EventListener/LoadDataContainerListener.php [hunk 1]
```diff
@@ -12,9 +12,6 @@
 
 class LoadDataContainerListener
 {
-    /**
-     * @Hook("loadDataContainer")
```
Comment (rabauss) [inline src/EventListener/LoadDataContainerListener.php:16]: was not used because it's registered in `config.php` and annotations are not supported with contao 4.4

File: src/EventListener/ReplaceInsertTagsListener.php [hunk 2]
```diff
@@ -22,9 +22,6 @@ public function __construct(MapManager $mapManager)
         $this->mapManager = $mapManager;
     }
 
-    /**
-     * @Hook("replaceInsertTags")
```
Comment (rabauss) [inline src/EventListener/ReplaceInsertTagsListener.php:26]: was not used because it's registered in config.php and annotations are not supported with contao 4.4

File: src/Manager/MapManager.php [hunk 3]
```diff
@@ -181,7 +181,7 @@ public function render(int $mapId, array $config = [])
 
         /** @var BeforeRenderMapEvent $event */
         /** @noinspection PhpParamsInspection */
-        $event = $this->eventDispatcher->dispatch(BeforeRenderMapEvent::NAME, new BeforeRenderMapEvent($template, $templateData, $map));
+        $event = $this->eventDispatcher->dispatch(new BeforeRenderMapEvent($template, $templateData, $map), BeforeRenderMapEvent::NAME, );
```
Comment (rabauss) [inline src/Manager/MapManager.php:184]:
```suggestion
        $event = $this->eventDispatcher->dispatch(new BeforeRenderMapEvent($template, $templateData, $map), BeforeRenderMapEvent::NAME);
```

File: src/Manager/MapManager.php [hunk 4]
```diff
@@ -200,7 +200,7 @@ public function renderMapObject(Map $map, ?int $mapId = null)
...
```
Comment (rabauss) [inline src/Manager/MapManager.php:203]:
```suggestion
        $event = $this->eventDispatcher->dispatch(new BeforeRenderMapEvent($template, $templateData, $map), BeforeRenderMapEvent::NAME);
```

**Review by rabauss — COMMENTED** [review]
Fix event dispatching

**Comment by Defcon0** [comment]
> Allow to use the bundle with php 8 and fix symfony 5 support.
> I'm not sure if we should drop the support for contao 4.4 - what do you think?

Good idea since we don't have that much contao 4.4 systems anymore and the lts support is dropped in some months. At the moment we don't have a project for symfony 5, so it would be cool if you could take a look whether there's more to fix for that.

Nevertheless this should be beta still since we need more testing for this change then. In addition, google maps isn't used that often in projects because of privacy concerns. This also doesn't make it easier to have a real life example.

So do you think, there might be more to fix for symfony 5? PHP 8 support is not too much of a problem in my experience. We just started our first php 8 project and so far it works well for most of our modules.

File: src/DataContainer/GoogleMap.php [hunk 5]
```diff
@@ -182,7 +182,7 @@ public function checkPermission()
                                 $arrModulep = \StringUtil::deserialize($objGroup->contao_google_maps_bundlep);
 
                                 if (\is_array($arrModulep) && \in_array('create', $arrModulep)) {
-                                    $arrModules = \StringUtil::deserialize($objGroup->contao_google_maps_bundles, true);
```
Comment (Defcon0) [inline src/DataContainer/GoogleMap.php:185]: I think the problem here is more that we use the compat layer. Could you try to replace

`\StringUtil::deserialize` by `StringUtil::deserialize` and `use Contao\StringUtil`?

File: src/Resources/translations/messages.de.yml [hunk 6]
```diff
@@ -1,2 +1,2 @@
 # labels
-huh.google_maps.label.routing: Routenplaner zu Ihrer Adresse:
\ No newline at end of file
+huh.google_maps.label.routing: 'Routenplaner zu Ihrer Adresse:'
```
Comment (Defcon0) [inline src/Resources/translations/messages.de.yml:2]: Unnecessary. Why did you add the apostrophes?

Comment (rabauss) [reply src/Resources/translations/messages.de.yml:2]: Well the inspector of phpstorm complained that it's not best practice without quotes!

Comment (Defcon0) [reply src/Resources/translations/messages.de.yml:2]: Mhm, I guess it's a question of taste (https://stackoverflow.com/questions/19109912/yaml-do-i-need-quotes-for-strings-in-yaml), so no problem.

File: src/DataContainer/GoogleMap.php (hunk 5 above)
Comment (rabauss) [reply src/DataContainer/GoogleMap.php:185]: I added some not all (because I did not want to get to many changes!) definitions from https://github.com/contao/easy-coding-standard/blob/3.x/config/set/contao.php to find fixes ;-)

File: src/Resources/translations/messages.de.yml (hunk 6 above)
Comment (rabauss) [reply src/Resources/translations/messages.de.yml:2]: Yes you're right - I can also revert if you like!

**Comment by rabauss** [comment]
For me everything runs under Contao 4.12 and php 8 with the current code status of this PR!

**Comment by Defcon0** [comment]
OK, cool. Thanks for your work :-) Looks good to me. Didn't know too much about ECS, we'll integrate that in our workflows.

Are you ready with the PR so we can merge it?

File: src/Resources/translations/messages.de.yml (hunk 6 above)
Comment (Defcon0) [reply src/Resources/translations/messages.de.yml:2]: Yes, please.

**Comment by rabauss** [comment]
Yes all good for me! 

P.S. ecs has some nice features - my default config wold look like this:
https://github.com/bwein-net/contao-backend-customizer/blob/main/ecs.php

**Comment by Defcon0** [comment]
OK, I just released the new version :-)

File: src/Resources/translations/messages.de.yml (hunk 6 above)
Comment (rabauss) [reply src/Resources/translations/messages.de.yml:2]: Okay, my revert was to late - now the PR is with the quotes!

File: src/Manager/MapManager.php [hunk 7]
```diff
@@ -128,7 +128,7 @@ public function prepareMap(int $mapId, array $config = [], Collection $overlays
         $map->setVariable('map_'.$mapId.'_'.substr(md5(time().$mapId), 0, 8));
 
         // apply map config
-        $htmlId = $mapConfig->htmlId ?: 'map_canvas_'.uniqid();
+        $htmlId = $mapConfig->htmlId ?: 'map_canvas_'.uniqid('', true);
```
Comment (Defcon0) [inline src/Manager/MapManager.php:131]: I needed to revert that because the id then contained a dot leading to css selectors not working amymore.

Comment (rabauss) [reply src/Manager/MapManager.php:131]: okay, sorry for that! It was a security hint of the inspector
